src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IAnalyzeResultWriter.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.UpgradeAssistant
+{
+    public interface IAnalyzeResultWriter
+    {
+        void WriteAsync(Dictionary<string, ICollection<AnalyzeResult>> results, CancellationToken token);
```

Comment:
What's the dictionary here for? Without any docs, it's difficult to know what it is.

Also, in a public API, we shouldn't use `Dictionary<,>` but rather `IDictionary<,>`

Reply:
This does seem to warrant a custom interchange POCO of some sort to more cleanly define what the inputs are

Reply:
Added a POCO